### PR TITLE
Navigation us9

### PR DESCRIPTION
**Pull request by LHJE**
**Contributors**
Driver: Luke
Navigator: Priya

**What Does It Do**
Ensures an admin is sent to a 404 page if attempting to visit the cart or a merchant path.

### CODE REVIEW

File: app/controllers/cart/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Cart::BaseController < ApplicationController
```
Comment (priyapower) [inline app/controllers/cart/base_controller.rb:1]: WOOO! I'm so psyched how easily this worked. I still have a question about where authorization is actually occurring, but I think we can do a code review for this.

File: spec/features/navigation_spec.rb [hunk 2]
```diff
@@ -47,16 +47,42 @@
       end
     end
 
-    it "can redirect a user to 404 error if they try to access any path with /merchant" do
+    it "can redirect a regular user to 404 error if they try to access any path with /merchant" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:50]: So glad you caught this! Now our tests are much better - and actually confirm what was required

File: spec/features/navigation_spec.rb [hunk 3]
```diff
@@ -47,16 +47,42 @@
...
+    it "can redirect a regular user to 404 error if they try to access any path with /admin" do
+      fill_in :email, with: @user.email
+      fill_in  :password, with: @user.password
+      click_button "Login"
+
       visit "/admin"
       expect(page).to have_content("The page you were looking for doesn't exist.")
     end
 
+    it "can redirect an admin to 404 error if they try to access any path with /merchant" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:68]: This meets requirements for us9 admin users cannot see merchant pages

File: spec/features/navigation_spec.rb [hunk 4]
```diff
@@ -47,16 +47,42 @@
...
+      fill_in :email, with: @admin.email
+      fill_in  :password, with: @admin.password
+      click_button "Login"
+
+      visit "/merchants"
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+    end
+
+    it "can redirect an admin to 404 error if they try to access any path with /admin" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:77]: This meets requirements for us9 admin users cannot see cart pages

**Review by priyapower — COMMENTED** [review]
Looks epic!

File: app/controllers/cart/base_controller.rb (hunk 1 above)
Comment (LHJE) [reply app/controllers/cart/base_controller.rb:1]: Same here!

File: spec/features/navigation_spec.rb [hunk 5]
```diff
@@ -59,6 +59,24 @@
 
         expect(current_path).to eq("/register")
       end
+      
+      it "can redirect an admin to 404 error if they try to access any path with /merchant" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:63]: This was an intense git merge! Glad we worked through that together and learned that we need to do separate tests or setups for merges

**Pull request merged**